src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonConverter.cs

```diff
@@ -114,9 +113,16 @@ internal bool ShouldFlush(Utf8JsonWriter writer, ref WriteStack state)
 
         internal ConstructorInfo? ConstructorInfo { get; set; }
 
-        internal virtual bool RequiresDynamicMemberAccessors { get; }
+        /// <summary>
+        /// Used for hooking custom configuration to a newly created associated JsonTypeInfo instance.
+        /// </summary>
+        internal virtual void ConfigureJsonTypeInfo(JsonTypeInfo jsonTypeInfo, JsonSerializerOptions options) { }
 
-        internal virtual void Initialize(JsonSerializerOptions options, JsonTypeInfo? jsonTypeInfo = null) { }
+        /// <summary>
+        /// Additional reflection-specific configuration required by certain collection converters.
+        /// </summary>
+        [RequiresUnreferencedCode(JsonSerializer.SerializationUnreferencedCodeMessage)]
+        internal virtual void ConfigureJsonTypeInfoUsingReflection(JsonTypeInfo jsonTypeInfo, JsonSerializerOptions options) { }
```

Comment:
It's not clear from the comment why do you actually need both methods.. Why can't the certain converters just use the specific configuration in the other method?

Reply:
It's because of linker safety. Reusing the same method in source gen paths would introduce a static dependency to unreferenced code resulting in linker warnings.